cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -185,24 +185,55 @@ def _parameter_names_(self) -> AbstractSet[str]:
             )
         }
 
-    def _decompose_(self) -> 'cirq.OP_TREE':
-        result = self.circuit.unfreeze()
-        result = result.transform_qubits(lambda q: self.qubit_map.get(q, q))
+    def mapped_circuit(self, deep: bool = False) -> 'cirq.Circuit':
```

Comment:
Add a docstring.  It should also explain what deep is.

Reply:
Done and done.